docker/Dockerfile

```diff
@@ -1,19 +1,21 @@
-FROM --platform=linux/amd64 node:20.2-slim
-
-COPY docker/package.json ./
+FROM --platform=linux/amd64 node:20.2-slim AS deps
+COPY package.json ./
 RUN npm install
 
-WORKDIR /usr/src/app
-
-# copy our code and create build folder in container
-COPY public ./public
-COPY img ./img
-COPY build ./build
-
-COPY docker/server ./server
+FROM deps AS build
+COPY ../public ./public
+COPY ../img ./img
+COPY ../src ./src
+COPY ../tsconfig.json ./
+COPY ../.eslintrc.js ./
+RUN npm run build
 
+FROM --platform=linux/amd64 node:20.2-slim AS runner
+WORKDIR /usr/src/app
+COPY --from=build build ./build
```

Comment:
These all "build" names are making this line little confusing. I would change FROM deps AS build to builder. It will make it easier to understand.